src/zenml/integrations/huggingface/steps/accelerate_runner.py

```diff
@@ -65,100 +66,99 @@ def training_pipeline(some_param: int, ...):
+                    entrypoint, flavour="accelerate"
+                ) as (
+                    script_path,
+                    output_path,
+                ):
+                    commands = [str(script_path.absolute())]
+                    for k, v in kwargs.items():
+                        k = _cli_arg_name(k)
+                        if isinstance(v, bool):
+                            if v:
+                                commands.append(f"--{k}")
+                        elif type(v) in (list, tuple, set):
+                            for each in v:
+                                commands += [f"--{k}", f"{each}"]
+                        else:
+                            commands += [f"--{k}", f"{v}"]
+                    logger.debug(commands)
+
+                    parser = launch_command_parser()
+                    args = parser.parse_args(commands)
+                    for k, v in accelerate_launch_kwargs.items():
+                        if k in args:
+                            setattr(args, k, v)
+                        else:
+                            logger.warning(
+                                f"You passed in `{k}` as an `accelerate launch` argument, but it was not accepted. "
+                                "Please check https://huggingface.co/docs/accelerate/en/package_reference/cli#accelerate-launch "
+                                "to find out more about supported arguments and retry."
+                            )
+                    try:
+                        launch_command(args)
+                    except Exception as e:
+                        logger.error(
+                            "Accelerate training job failed... See error message for details."
+                        )
+                        raise RuntimeError(
+                            "Accelerate training job failed."
+                        ) from e
+                    else:
+                        logger.info(
+                            "Accelerate training job finished successfully."
+                        )
+                        return pickle.load(open(output_path, "rb"))
+
+            return cast(F, inner)
+
+        try:
+            get_pipeline_context()
+        except RuntimeError:
+            pass
+        else:
+            raise RuntimeError(
+                f"`{run_with_accelerate.__name__}` decorator cannot be used "
+                "in a functional way with steps, please apply decoration "
+                "directly to a step instead. This behavior will be also "
+                "allowed in future, but now it faces technical limitations.\n"
+                "Example (allowed):\n"
+                f"@{run_with_accelerate.__name__}(...)\n"
+                f"def {step_function.name}(...):\n"
+                "    ...\n"
+                "Example (not allowed):\n"
+                "def my_pipeline(...):\n"
+                f"    run_with_accelerate({step_function.name},...)(...)\n"
```

Comment:
I am not too familiar with `accelerate` myself, so I might be missing something here, but if this `RuntimeError` is true, we might want to change the docstring of the `run_with_accelerate` function, as it seems like the example that is given there is not allowed anymore.

Reply:
Exactly the same is written on L143 🙂 `Example (not allowed):\n`

Reply:
I was talking about L55 😄 

```python
@pipeline
def training_pipeline(some_param: int, ...):
    run_with_accelerate(training_step, num_processes=4)(some_param, ...)
```

Isn't this the not-allowed case?

Reply:
Yeah, indeed. Good catch 👍🏼